pycytominer/cyto_utils/__init__.py

```diff
@@ -40,4 +40,14 @@
     aggregate_fields_count,
     aggregate_image_features,
 )
-from .collate import collate
+
+# Conditionally import collate function depending on if optional dependencies installed
+try:
+    from .collate import collate
+except ImportError as error:
+
+    def collate(error=error, *args, **kwargs):
```

Comment:
Instead of redefining `collate` here, consider moving the `cytominer_database` import within `collate.py` to a localized function import where it's used. This could grant a similar functionality without modifying the `__init__.py` or implementing redefinition of `collate`. [We already do this with the SingleCells class](https://github.com/cytomining/pycytominer/blob/main/pycytominer/cyto_utils/collate.py#L73) so there's some precedence to the style of the approach here.

When it comes to Pythonic approach, this is totally single opinion and not necessarily canon 🙂 . Generally I feel the above might be "simple over complex", avoid broad-reaching changes within the `__init__.py` (keeping impact minimized), and could help with avoiding future code decay by aligning logic closer to implementation.

Reply:
This suggestion makes a lot of sense to me. I've seen localized imports before, but didn't really have a sense of their purpose. Keeping the imports of optional dependencies restricted to the functions that need them seems like an excellent approach.